src/euphorie/client/model.py

```diff
@@ -801,20 +802,42 @@ def is_archived(self):
             return False
         return archived <= localized_now()
 
+    @property
+    def last_locking_event(self):
+        """Return the last event relative to locking"""
+        query = (
+            Session.query(SessionEvent)
+            .filter(
+                SessionEvent.action.in_(("lock_set", "lock_reset", "lock_unset")),
+                SessionEvent.session_id == self.id,
+            )
+            .order_by(SessionEvent.time.desc())
+        )
+        return query.first()
+
     @property
     def is_locked(self):
+        """Check if the session is locked."""
         consultancy = self.consultancy
-        if not consultancy:
+        if consultancy and consultancy.status == "validated":
+            return True
```

Comment:
```suggestion
```

I'd suggest explicitly setting the lock in the consultancy view so that you don't need another case distinction here.

Alternatively add the validated event to the list of locking events (but that seems like a violation of separation of concerns).

Reply:
This has indeed to change, but not by setting the lock but by checking if a "validate sesssion event" has been recorded.
Don't you agree?

Reply:
As I said, that seems like a violation of separation of concerns. The locking code should not have to check the validation status of a session. Also, in the long run an admin should be able to unlock a session even if it is validated.

Reply:
We can tweak this later though.